### PR TITLE
Add new role "Technical Support - LJ" (API, User management for LG)

### PR DESCRIPTION
**Pull request by guri-dev**
## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## UI Accessibility Checklist

_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

- [ ] Markup uses semantic HTML?
- [ ] Additional context added through `aria-roles` and `aria-labels`?
- [ ] Checked
      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?
- [ ] Usable with a keyboard and navigable in a logical order?
- [ ] Usable using a screen reader (e.g. Voiceover for Mac) with enough context for the user?

## General Checklist

- [ ] ✅ Provide tests for your changes where relevant
- [ ] 📝 Use descriptive commit messages
- [ ] 📗 Update any related documentation and include any relevant screenshots
- [ ] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size
- [ ] 🗂️ Move your relevant story/task to the _Ready for Code Review_ state

### CODE REVIEW

File: app/frontend/i18n/i18n.ts [hunk 1]
```diff
@@ -1491,6 +1491,7 @@ const options = {
             review_manager: "review manager",
             reviewer: "reviewer",
             super_admin: "super admin",
+            technical_support_local_jurisdiction: "Technical Support - LJ",
```
Comment (bkmaibach) [inline app/frontend/i18n/i18n.ts:1494]: you should probably just shorten this to technical_support for the sake of brevity and sanity

File: app/models/user.rb [hunk 2]
```diff
@@ -120,7 +121,7 @@ def search_data
   def invitable_roles
     case role
     when "super_admin"
-      %w[reviewer review_manager super_admin regional_review_manager]
+      %w[reviewer review_manager super_admin regional_review_manager technical_support_local_jurisdiction]
     when "review_manager", "regional_review_manager"
       %w[reviewer review_manager]
```
Comment (bkmaibach) [inline app/models/user.rb:126]: if you want review managers to be able to invite these kinds of users, add it here

**Review by bkmaibach — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (guri-dev) [reply app/models/user.rb]: This has been updated in latest commit

File: app/frontend/i18n/i18n.ts (hunk 1 above)
Comment (guri-dev) [reply app/frontend/i18n/i18n.ts]: This has been updated

File: app/models/user.rb [hunk 3]
```diff
@@ -221,7 +222,7 @@ def confirmed_user_has_fields
   end
 
   def jurisdiction_must_belong_to_correct_roles
-    if jurisdictions.any? && !review_staff?
+    if jurisdictions.any? && !review_staff? && !technical_support?
```
Comment (bkmaibach) [inline app/models/user.rb:225]: What I would do is add a new computed value jurisdiction_staff?
check everywhere review_staff? is used, and determine if it should be replaced with jurisdiction_staff?

File: app/frontend/types/enums.ts [hunk 4]
```diff
@@ -63,6 +63,7 @@ export enum EUserRoles {
   reviewManager = "review_manager",
   reviewer = "reviewer",
   superAdmin = "super_admin",
+  technicalSupport = "technical_support",
```
Comment (bkmaibach) [inline app/frontend/types/enums.ts:66]: Does this make this type invitable automatically?

**Review by bkmaibach — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (guri-dev) [reply app/models/user.rb:225]: @smithskye  I made a new method that will be responsible for verifying the user role wherever we needed in the app.

File: app/frontend/types/enums.ts (hunk 4 above)
Comment (guri-dev) [reply app/frontend/types/enums.ts:66]: @smithskye  When we invite a new user by selecting the user role from the drop-down here, it automatically invites the user with the Technical Support role.

image

**Review by smithskye — APPROVED** [review]
Reviewed, What I am able to for this role.

File: app/models/user.rb [hunk 5]
```diff
@@ -252,8 +255,12 @@ def confirmed_user_has_fields
     end
   end
 
+  def jurisdiction_staff?
+    !review_staff? && !technical_support?
+  end
+
   def jurisdiction_must_belong_to_correct_roles
```
Comment (bkmaibach) [inline app/models/user.rb:262]: The way this is set up effectively gives the technical support the same permissions as a submitter, except belongs to a jurisdiction. The tech support is not currently able to access jurisdiction configuration, users, etc.

Comment (guri-dev) [reply app/models/user.rb:266]: @bkmaibach  This has updated to check only review_staff role, handled technical_support in different method.

File: app/models/user.rb [hunk 6]
```diff
@@ -138,6 +140,10 @@ def manager?
     review_manager? || regional_review_manager?
   end
 
+  def technical_support?
+    technical_support?
```
Comment (bkmaibach) [inline app/models/user.rb:144]: this method should already exist automatically due to the enum

File: app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx [hunk 7]
```diff
@@ -52,63 +54,77 @@ export const ConfigurationManagementScreen = observer(function ConfigurationMana
...
-                  title={t(`${i18nPrefix}.globalFeatureAccess.title`)}
-                  description={t(`${i18nPrefix}.globalFeatureAccess.description`)}
-                  linkText={t("ui.edit")}
-                  icon={<SlidersHorizontal size="24px" color="var(--chakra-colors-text-link)" />}
-                  href="global-feature-access"
-                  h="full"
-                />
-              </GridItem>
-            </Grid>
+            {currentUser.role == EUserRoles.technicalSupport && (
```
Comment (bkmaibach) [inline app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx:57]: Not a fan of this approach. Why not give the tech support its own home screen like the rest of the roles?

File: app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+export const TechnicalSupportHomeScreen = observer(({ ...rest }: IHomeScreenProps) => {
+  
+  const {
+    userStore: {
+      currentUser: { jurisdiction },
+    },
+  } = useMst()
+  const navigate = useNavigate()
+
+  useEffect(() => {
```
Comment (bkmaibach) [inline app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx:17]: Antipattern
Please create a role specific component

File: app/policies/jurisdiction_policy.rb [hunk 9]
```diff
@@ -20,7 +20,7 @@ def create?
   end
 
   def update?
-    user.super_admin? || (user.staff? && user.jurisdictions.find(record.id))
+    user.super_admin? || (user.staff? && user.jurisdictions.find(record.id)) || user.technical_support
```
Comment (bkmaibach) [inline app/policies/jurisdiction_policy.rb:23]: You want to allow tech support to modify any jurisdiction, even if they are not assigned to it?

File: app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx (hunk 8 above)
Comment (bkmaibach) [reply app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx:17]: Instead of using an effect to go to another role's component, and then making exceptions inside of that component, wouldnt it just be better to have

review manager -> review manager screen
reviewer -> reviewer screen
tech support -> tech support screen

etc?

File: app/models/user.rb (hunk 5 above)
Comment (bkmaibach) [reply app/models/user.rb:268]: Is tech support not considered jurisdiciton staff?
is this accurate naming?

File: app/policies/jurisdiction_policy.rb (hunk 9 above)
Comment (bkmaibach) [reply app/policies/jurisdiction_policy.rb:23]: Staff could use this to invite themself as tech support, and then give themselves access to update any jurisdiction, thus defeating the purpose of jurisdiciton membership in the first place

**Review by bkmaibach — COMMENTED** [review]
We may need an excel spreadsheet or some report on all policies, and where tech support, review staff, and jurisdiciton membership applies in each case before, and after.

File: app/models/user.rb (hunk 5 above)
Comment (guri-dev) [reply app/models/user.rb:268]: @bkmaibach  No, tech support is not a part of review staff because review staff can access the design templates, and Tech support should not.

File: app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx (hunk 7 above)
Comment (guri-dev) [reply app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx]: @bkmaibach  Replaced this code with a new component for Technical Support user role

File: app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx (hunk 8 above)
Comment (guri-dev) [reply app/frontend/components/domains/home/technical-support/technical-support-home-screen.tsx]: @bkmaibach  It has been removed/replaced with a new Technical Support screen for new user role.

File: app/models/user.rb (hunk 5 above)
Comment (bkmaibach) [reply app/models/user.rb:268]: This is naming is confusing and not very straightforward

File: app/models/user.rb [hunk 10]
```diff
@@ -252,8 +261,12 @@ def confirmed_user_has_fields
...
   end
 
+  def jurisdiction_staff?
+    !review_staff?
+  end
+
   def jurisdiction_must_belong_to_correct_roles
-    if jurisdictions.any? && !review_staff?
+    if jurisdictions.any? && jurisdiction_staff? && !technical_support?
       errors.add(:jurisdictions, :reviewers_only)
```
Comment (bkmaibach) [inline app/models/user.rb:270]: Why cant this be left as !review_staff?

File: app/policies/jurisdiction_policy.rb [hunk 11]
```diff
@@ -25,7 +25,8 @@ def update?
 
   def update_external_api_enabled?
     user.super_admin? ||
-      (user.manager? && user.jurisdictions.find(record.id) && !record.g_off?)
+      (user.manager? && user.jurisdictions.find(record.id) && !record.g_off?) ||
+      (user.technical_support? && user.jurisdictions.find(record.id) && !record.g_off?)
```
Comment (bkmaibach) [inline app/policies/jurisdiction_policy.rb:29]: repetitive logic

**Review by bkmaibach — CHANGES_REQUESTED** [review]

File: app/policies/jurisdiction_policy.rb (hunk 11 above)
Comment (guri-dev) [reply app/policies/jurisdiction_policy.rb]: @bkmaibach  This has been resolved

File: app/models/user.rb (hunk 10 above)
Comment (guri-dev) [reply app/models/user.rb:270]: @bkmaibach  I updated this changes as suggested

File: app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx [hunk 12]
```diff
@@ -52,63 +54,50 @@ export const ConfigurationManagementScreen = observer(function ConfigurationMana
...
+                    description={t(`${i18nPrefix}.users.description`)}
+                    linkText={t("ui.edit")}
+                    icon={<Users size={24} />}
+                    href={`/jurisdictions/${currentJurisdiction.slug}/users`}
+                    h="full"
+                    disableForSandbox
+                  />
+                </GridItem>
+                <GridItem>
+                  <HomeScreenBox
```
Comment (bkmaibach) [inline app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx:91]: This needs to be rebased on the global feature access stuff I think? this has the old feature access stuff it looks like

Comment (guri-dev) [reply app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx:91]: Yes, it's checkout from the older version.

**Review by bkmaibach — COMMENTED** [review]
image
Currently getting this when navigating to users list

**Comment by guri-dev** [comment]
@bkmaibach, I just updated the PR regarding the user access issue mentioned above. Please review.

File: app/policies/user_policy.rb [hunk 13]
```diff
@@ -29,7 +29,7 @@ def super_admins?
 
   def search_jurisdiction_users?
     return true if user.super_admin? && (record.manager? || record.super_admin?)
-    user.manager? && record_in_users_jurisdictions?
+    (user.manager? || user.technical_support?) && record_in_users_jurisdictions?
```
[the remark this answers is not in the document]
Comment (bkmaibach) [reply app/policies/user_policy.rb:32]: Looking back at this it looks like I might have messed up my screenshot, but maybe just check out user search as TS

[the remark this answers is not in the document]
Comment (guri-dev) [reply app/policies/user_policy.rb:32]: image
I tested the user search as a TS and it works as expected

File: app/controllers/api/external_api_keys_controller.rb [hunk 14]
```diff
@@ -4,7 +4,7 @@ class Api::ExternalApiKeysController < Api::ApplicationController
   def index
     # Only authorized to query own jurisdiction for review managers
     if (
-         current_user.review_manager? || current_user.regional_review_manager?
+         current_user.review_manager? || current_user.regional_review_manager?  || current_user.technical_support?
```
[the remark this answers is not in the document]
Comment (guri-dev) [reply app/controllers/api/external_api_keys_controller.rb:7]: @bkmaibach  No because the TS user does not have all the permissions similar to the review staff.

File: app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx [hunk 15]
```diff
@@ -52,63 +54,50 @@ export const ConfigurationManagementScreen = observer(function ConfigurationMana
...
-                  title={t(`${i18nPrefix}.globalFeatureAccess.title`)}
-                  description={t(`${i18nPrefix}.globalFeatureAccess.description`)}
-                  linkText={t("ui.edit")}
-                  icon={<SlidersHorizontal size="24px" color="var(--chakra-colors-text-link)" />}
-                  href="global-feature-access"
-                  h="full"
-                />
-              </GridItem>
-            </Grid>
+              <Grid templateColumns="repeat(2, 1fr)" gap={6}>
```
[the remark this answers is not in the document]
Comment (guri-dev) [reply app/frontend/components/domains/home/review-manager/configuration-management-screen/index.tsx]: @bkmaibach  This has been fixed

File: app/frontend/components/domains/home/technical-support/index.tsx [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+  if (!currentUser?.jurisdiction) return <ErrorScreen error={new Error(t("errors.fetchJurisdiction"))} />
+  const jurisdiction = currentUser.jurisdiction
+
+  return (
+    <Container maxW="container.md" py={16} as="main">
+      <Flex direction="column" align="center" w="full">
+        <Heading as="h1" mb={8}>
+          {jurisdiction.name}
+        </Heading>
+        <Flex direction="column" align="center" w="full" gap={6}>
```
[the remark this answers is not in the document]
Comment (guri-dev) [reply app/frontend/components/domains/home/technical-support/index.tsx:25]: @bkmaibach  Link has been added to the home screen.

File: app/frontend/models/user.ts [hunk 17]
```diff
@@ -121,6 +124,8 @@ export const UserModel = types
         newRole = EUserRoles.reviewer
       } else if (self.role === EUserRoles.reviewer) {
         newRole = EUserRoles.reviewManager
+      } else if (self.role === EUserRoles.technicalSupport) {
```
Comment (bkmaibach) [inline app/frontend/models/user.ts:127]: Are we sure we want to be able to change a tech support to a review/manager, but not back?

**Review by bkmaibach — APPROVED** [review]
